Mathlib/GroupTheory/Subgroup/MulOpposite.lean

```diff
@@ -20,44 +21,61 @@ variable {G : Type*} [Group G]
 
 namespace Subgroup
 
+/-- pull a subgroup back to an opposite subgroup along `unop`-/
+@[to_additive (attr := simps)
+"pull an additive subgroup back to an opposite additive subgroup along `unop`"]
+def op (H : Subgroup G) : Subgroup Gᵐᵒᵖ where
+  carrier := MulOpposite.unop ⁻¹' (H : Set G)
+  one_mem' := H.one_mem
+  mul_mem' ha hb := H.mul_mem hb ha
+  inv_mem' := H.inv_mem
+
+/-- pull an opposite subgroup back to a subgroup along `op`-/
+@[to_additive (attr := simps)
+"pull an opposite additive subgroup back to an additive subgroup along `op`"]
+def unop (H : Subgroup Gᵐᵒᵖ) : Subgroup G where
+  carrier := MulOpposite.op ⁻¹' (H : Set Gᵐᵒᵖ)
+  one_mem' := H.one_mem
+  mul_mem' := fun ha hb => H.mul_mem hb ha
+  inv_mem' := H.inv_mem
+
 /-- A subgroup `H` of `G` determines a subgroup `H.opposite` of the opposite group `Gᵐᵒᵖ`. -/
-@[to_additive "An additive subgroup `H` of `G` determines an additive subgroup `H.opposite` of the
+@[to_additive (attr := simps) "An additive subgroup `H` of `G` determines an additive subgroup `H.opposite` of the
  opposite additive group `Gᵃᵒᵖ`."]
-def opposite : Subgroup G ≃ Subgroup Gᵐᵒᵖ
-    where
-  toFun H :=
-    { carrier := MulOpposite.unop ⁻¹' (H : Set G)
-      one_mem' := H.one_mem
-      mul_mem' := fun ha hb => H.mul_mem hb ha
-      inv_mem' := H.inv_mem }
-  invFun H :=
-    { carrier := MulOpposite.op ⁻¹' (H : Set Gᵐᵒᵖ)
-      one_mem' := H.one_mem
-      mul_mem' := fun ha hb => H.mul_mem hb ha
-      inv_mem' := H.inv_mem }
+def opEquiv : Subgroup G ≃ Subgroup Gᵐᵒᵖ where
+  toFun := op
+  invFun := unop
   left_inv _ := SetLike.coe_injective rfl
   right_inv _ := SetLike.coe_injective rfl
-#align subgroup.opposite Subgroup.opposite
-#align add_subgroup.opposite AddSubgroup.opposite
+#align subgroup.opposite Subgroup.opEquiv
+#align add_subgroup.opposite AddSubgroup.opEquiv
+
+@[to_additive (attr := simp, nolint simpNF)] lemma opposite_toSubmonoid (H : Subgroup G) :
+    (opEquiv H).toSubmonoid = Submonoid.opEquiv H.toSubmonoid :=
+  rfl
```

Comment:
```suggestion
@[to_additive (attr := simp, nolint simpNF)] lemma opposite_toSubmonoid (H : Subgroup G) :
    H.op.toSubmonoid = H.toSubmonoid.op :=
  rfl
```
as I wouldn't expect your spelling to be simp-normal. The name and position of the lemma should also change, as should the `symm` version below.

Reply:
done